extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/report/MSQResultsReport.java

```diff
@@ -63,7 +68,7 @@ static MSQResultsReport fromJson(
   }
 
   @JsonProperty("signature")
-  public RowSignature getSignature()
+  public List<ColumnAndType> getSignature()
```

Comment:
Nice. I guess we would not require any json changes with this. 
+1

Reply:
Yep that's the idea. The JSON looks the same.